src/object/srv_obj_migrate.c

```diff
@@ -2573,43 +2582,33 @@ ds_migrate_fini_one(uuid_t pool_uuid, uint32_t ver)
 
 	migrate_pool_tls_put(tls); /* destroy */
 
-	D_DEBUG(DB_TRACE, "migrate fini one ult "DF_UUID"\n", pool_uuid);
-}
-
-struct migrate_abort_arg {
-	uuid_t	pool_uuid;
-	uint32_t version;
-};
-
-int
-migrate_fini_one_ult(void *data)
-{
-	struct migrate_abort_arg *arg = data;
-
-	ds_migrate_fini_one(arg->pool_uuid, arg->version);
-
+	D_INFO("migrate fini one ult "DF_UUID"\n", DP_UUID(arg->pool_uuid));
 	return 0;
 }
 
-/* Abort the migration */
+/* stop the migration */
 void
-ds_migrate_abort(uuid_t pool_uuid, unsigned int version)
+ds_migrate_stop(struct ds_pool *pool, unsigned int version)
 {
 	struct migrate_pool_tls *tls;
-	struct migrate_abort_arg arg;
+	struct migrate_stop_arg arg;
 	int			 rc;
 
-	tls = migrate_pool_tls_lookup(pool_uuid, version);
+	tls = migrate_pool_tls_lookup(pool->sp_uuid, version);
 	if (tls == NULL)
 		return;
 
-	uuid_copy(arg.pool_uuid, pool_uuid);
+	uuid_copy(arg.pool_uuid, pool->sp_uuid);
 	arg.version = version;
 	rc = dss_thread_collective(migrate_fini_one_ult, &arg, 0);
 	if (rc)
-		D_ERROR("migrate abort: %d\n", rc);
+		D_ERROR(DF_UUID" migrate stop: %d\n", DP_UUID(pool->sp_uuid), rc);
 
+	if (tls->mpt_opc == RB_OP_REINT)
+		pool->sp_reintegrating--;
+	migrate_pool_tls_put(tls);
```

Comment:
One put is for tls_lookup(), the other put is for migrate_pool_tls_create_one()? Better to add some comment.